optuna/study/_multi_objective.py

```diff
@@ -91,26 +91,31 @@ def _fast_non_dominated_sort(
                 len(penalty), len(objective_values)
             )
         )
-    nondomination_rank = np.zeros(len(objective_values), dtype=int)
+    nondomination_rank = np.full(len(objective_values), -1)
     is_nan = np.isnan(penalty)
+    n_below = n_below or len(objective_values)
 
     # First, we calculate the domination rank for feasible trials.
     is_feasible = np.logical_and(~is_nan, penalty <= 0)
-    ranks, feasible_bottom_rank = _calculate_nondomination_rank(
+    ranks, bottom_rank = _calculate_nondomination_rank(
```

Comment:
If we define `nondomination_rank` as:
```python
nondomination_rank = np.full(len(objective_values), n_below + 1)
```

`bottom_rank` becomes `bottom_rank = np.max(ranks)`.
Note that if `np.max(bottom_rank) = n_below + 1`, the processes hereafter simply define each `nondomination_rank` as `n_below + <positive_integer>`, so they will be ignored.

Reply:
I totally agree what you say but it makes this PR even larger. Can I split the task as a follow-up and resolve your comment in another PR?

Reply:
Sure, then could you please add an inline comment for this?

Reply:
The suggestion is a little bit complicated, so I remarked the comment on https://github.com/optuna/optuna/issues/5089